### DISCUSSION TITLE
Threejs Object local clipping

### DISCUSSION THREAD

**Discussion by paulsoning** [heremaps/harp.gl#2326] [Q&A] [answered]
Hey there,

I would like to use local clipping of 3D Object that I inserted to the map.
The clipping plane is working, but unfortunately when I move the map the the clipping plane "stays" in the same place

Could someone point me in a right direction how to solve this problem.
```
        // ***** Clipping planes: *****
        const local = new THREE.Plane(new THREE.Vector3(0, -1, 0), 0.8);

        // Geometry
        const material = new THREE.MeshPhongMaterial({
            color: 0x80EE10,
            shininess: 100,
            side: THREE.DoubleSide,
            // ***** Clipping setup (material): *****
            clippingPlanes: [local],
            clipShadows: true,
        });

        const geometry = new THREE.TorusKnotBufferGeometry(0.4, 0.08, 95, 20);
        const mesh = new THREE.Mesh(geometry, material);
        mesh.scale.set(45, 45, 45);
        mesh.castShadow = true;
        
        group.add(mesh);
        group.anchor = figureGeoPosition;
        group.overlay = true;
        map.mapAnchors.add(group);
```
https://user-images.githubusercontent.com/11157695/138443216-3279248c-00b5-4c32-8e1b-d46935c90623.mov

Br,
Marek

**Comment by nzjony** [comment, marked as answer]
hi @paulsoning , we render objects relative to the camera, see: https://github.com/heremaps/harp.gl/blob/bd9d86e794919c21fbba1a4def1a85db5640d2e8/%40here/harp-mapview/lib/TileObjectsRenderer.ts#L66

As such, you need to transform the plane based on the position of the object, see: https://stackoverflow.com/questions/45611247/local-clipping-of-planes-based-on-local-axis

Something like:

```
        const updateListener = () => {
            // update position of plane based on the object's position
        };
        mapView.addEventListener(MapViewEventNames.Render, updateListener);
 ```

**Reply by paulsoning** [reply]
thanks, this works!

**Reply by nzjony** [reply]
@paulsoning , look forward to see what cool stuff you make!
